src/Shared/FileMatcher.cs

```diff
@@ -2277,6 +2300,53 @@ private static int SkipSlashes(string aString, int startingIndex)
             return index;
         }
 
+        /// <summary>
+        /// Returns true if drive enumerating wildcard patterns are detected using the directory and wildcard parts.
+        /// </summary>
+        private static bool IsDriveEnumeratingWildcardPattern(string directoryPart, string wildcardPart)
+        {
+            int directoryPartLength = directoryPart.Length;
+            int wildcardPartLength = wildcardPart.Length;
+            if (directoryPartLength >= 3 &&                                                                  // Handles <drive letter>:<slashes>** cases
+                wildcardPartLength >= 2 &&
+                IsDrivePatternWithoutSlash(directoryPart[0], directoryPart[1]))
+            {
+                for (int i = 2; i < directoryPartLength-1; i++)
+                {
+                    if (!FileUtilities.IsAnySlash(directoryPart[i]))
+                    {
+                        return false;
+                    }
+                }
+
+                return IsFullFileSystemScan(directoryPart[directoryPartLength - 1], wildcardPart[0], wildcardPart[1]);
+            }
+            else if (directoryPartLength == 1 &&                                                             // Handles \** case
+                     wildcardPartLength >= 2 &&
+                     IsFullFileSystemScan(directoryPart[0], wildcardPart[0], wildcardPart[1]))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Returns true if given characters follow a drive pattern without the slash (ex: C:).
+        /// </summary>
+        private static bool IsDrivePatternWithoutSlash(char firstValue, char secondValue)
+        {
+            return (IsValidDriveChar(firstValue)) && (secondValue == ':');
```

Comment:
I think there's a static character you can use here.

Yeah there's an `internal static readonly char[] ColonChar = { ':' };` in `Shared/Constants.cs`